static/js/entry/zendesk_widget.js

```diff
@@ -47,65 +47,104 @@ zE(function() {
     zE.identify(identity);
   }
 
-  // trigger onZendeskIFrameExists at the appropriate time
+  setupZendeskCallbacks();
+});
+
+const zendeskCallbacks = {
+  // supports the following functions:
+  //   launcherExists: runs when the launcher iframe exists on the page
+  //   launcherLoaded: runs when the launcher iframe's content is loaded
+  //   ticketSubmissionFormExists: runs when the submission form iframe exists
+  //   ticketSubmissionFormLoaded: runs when the submission form is loaded
+  //   npsExists: runs when the NPS iframe exists on the page
+  //   npsLoaded: runs when the NPS iframe is loaded
+  //   ipmExists: runs when the IPM iframe exists on the page
+  //   ipmLoaded: runs when the IPM iframe is loaded
+  // NPS = Net Promoter Score
+  // IPM = In-Product Message (Zendesk Connect)
+
+  launcherLoaded: () => {
+    const iframe = document.querySelector("iframe.zEWidget-launcher");
+    const btn = iframe.contentDocument.querySelector(".Button--launcher");
+
+    const regularBackgroundColor = "rgba(0, 0, 0, .14)";
+    const hoverBackgroundColor = window.getComputedStyle(btn).backgroundColor;
+    // We need to set a new background color, and unfortunately,
+    // the existing background color is set with "!important".
+    // As a result, the only way to override this existing color is to
+    // *also* use "!important".
+    const setHover = () => {
+      btn.style.setProperty(
+        "background-color", hoverBackgroundColor, "important"
+      );
+    };
+    const unsetHover = () => {
+      btn.style.setProperty(
+        "background-color", regularBackgroundColor, "important"
+      );
+    };
+    btn.onmouseenter = setHover;
+    btn.onmouseleave = unsetHover;
+    unsetHover();
+
+    // prepopulate ticket submission form
+    if (SETTINGS.program) {
+      const programSlug = SETTINGS.program.slug;
+      btn.onclick = () => {
+        setTimeout(() => {
+          const ticketIFrame = document.querySelector("iframe.zEWidget-ticketSubmissionForm");
+          let select = ticketIFrame.contentDocument.querySelector("select");
+          const optionValues = _.map(select.options, "value");
+          if (optionValues.includes(programSlug)) {
+            select.value = programSlug;
+          }
+        }, 100);
+      };
+    }
+  }
+};
+
+const setupZendeskCallbacks = () => {
+  const zendeskIFrames = ["launcher", "ticketSubmissionForm", "nps", "ipm"];
```

Comment:
I had thought that I would need to use the `ticketSubmissionFormLoaded` function, to modify the submission form as soon as it loaded. However, it appears that Zendesk's Javascript re-renders that form when the launcher button is clicked, so my modifications got clobbered by the re-render. Once I restructured the code so that I could define these arbitrary lifecycle functions, though, I figured it was an improvement over what was there before, so I left it as-is for the pull request.